rewrite-maven/src/main/java/org/openrewrite/maven/RemoveDuplicateDependencies.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.tree.Xml;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.UnaryOperator;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class RemoveDuplicateDependencies extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Remove duplicate Maven dependencies";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Removes duplicated dependencies in the <dependencies> section of the pom.xml.";
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(2);
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new RemoveDuplicateDependenciesVisitor();
+    }
+
+    private static class RemoveDuplicateDependenciesVisitor extends MavenIsoVisitor<ExecutionContext> {
+        private final XPathMatcher DEPENDENCIES_MATCHER = new XPathMatcher("/project/dependencies");
+        private static final XPathMatcher MANAGED_DEPENDENCIES_MATCHER = new XPathMatcher("/project/dependencyManagement/dependencies");
+
+        @Override
+        public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
+            if (isDependenciesTag() || isManagedDependenciesTag()) {
+                List<Xml.Tag> children = tag.getChildren();
+                List<Xml.Tag> newChildren = ListUtils.map(children, new DuplicatesRemover());
+                return children.equals(newChildren) ? tag : tag.withContent(newChildren);
```

Comment:
I noticed that the two recipes `RemoveExclusion` and `RemoveRedundantDependencyVersions` use the same basic pattern, but instead do it in one line like this:
```java
return tag.withContent(ListUtils.map(tag.getChildren(), ...));
```
When I copied that then I noticed that `Xml.Tag#withContent()` does a `this.content == content` comparison to determine if it should do an early return. But since `Xml.Tag#getChildren()` always returns a new list, this check will basically never be true. So in order to avoid a second cycle I did it this way to use `Object#equals()` as the early return check.

Possibly it would make sense to look into this. Maybe the return value for `getChildren()` should be cached (requiring cache invalidation logic) or maybe `withContent()` should do an `Object#equals()` on the lists.

Reply:
I realized that the `tag.withContent(ListUtils.map(tag.getChildren(), ...))` is flawed, because it will not preserve comments. So I use `Xml.Tag#getContent()` instead.